rust/windows-client/src-tauri/src/client.rs

```diff
@@ -0,0 +1,69 @@
+use anyhow::Result;
+use clap::Parser;
+use cli::CliCommands as Cmd;
+
+mod cli;
+mod debug_commands;
+mod device_id;
+#[cfg(target_family = "unix")]
+mod gui {
+    use super::*;
+
+    pub enum ControllerRequest {
+        SignIn,
+    }
+
+    pub(crate) fn run(_: GuiParams) -> Result<()> {
+        // The Ubuntu CI runner doesn't have gdk and some other Tauri deps installed, so it fails unless we stub out the GUI.
+        panic!("The Tauri GUI isn't implemented for Linux.");
+    }
+}
+#[cfg(target_os = "windows")]
+mod gui;
+mod local_webserver;
+// Relies on some types from Tauri
+#[cfg(target_os = "windows")]
+mod settings;
+
+/// Prevents a problem where changing the args to `gui::run` breaks static analysis on non-Windows targets, where the gui is stubbed out
+#[allow(dead_code)]
+pub(crate) struct GuiParams {
+    /// The URL of an incoming deep link from a web browser
+    deep_link: Option<String>,
```

Comment:
Why would it ever be `None`?

Reply:
Also if this is a valid url we should use [Url](https://docs.rs/url/latest/url/struct.Url.html)

Reply:
I removed this completely in 6521cd88 since it only matters if we're starting the app from a deep link, it's not needed if we're only catching deep links once running.